frontend/dockerfile/dockerfile_test.go

```diff
@@ -3472,6 +3473,55 @@ RUN sh -c "[ $(cat /test5/foo) = 'hello' ]"
 	require.NoError(t, err)
 }
 
+func testAddURLChmod(t *testing.T, sb integration.Sandbox) {
+	skipDockerd(t, sb)
+	f := getFrontend(t, sb)
+	f.RequiresBuildctl(t)
+
+	resp := httpserver.Response{
+		Etag:    identity.NewID(),
+		Content: []byte("content1"),
+	}
+	server := httpserver.NewTestServer(map[string]httpserver.Response{
+		"/foo": resp,
+	})
+	defer server.Close()
+
+	dockerfile := []byte(fmt.Sprintf(`
+FROM busybox AS build
+ADD --chmod=644 %[1]s /tmp/foo1
+ADD --chmod=755 %[1]s /tmp/foo2
+ADD --chmod=0413 %[1]s /tmp/foo3
+RUN stat -c "%%04a" /tmp/foo1 >> /dest && \
+	stat -c "%%04a" /tmp/foo2 >> /dest && \
+	stat -c "%%04a" /tmp/foo3 >> /dest
+
+FROM scratch
+COPY --from=build /dest /dest
+`, server.URL+"/foo"))
+
+	dir, err := tmpdir(
+		fstest.CreateFile("Dockerfile", dockerfile, 0600),
+	)
+	require.NoError(t, err)
+	defer os.RemoveAll(dir)
+
+	args, trace := f.DFCmdArgs(dir, dir)
+	defer os.RemoveAll(trace)
+
+	destDir, err := tmpdir()
+	require.NoError(t, err)
+	defer os.RemoveAll(destDir)
+
+	cmd := sb.Cmd(args + fmt.Sprintf(" --exporter=local --exporter-opt output=%s", destDir))
```

Comment:
Sorry, missed this on the first review. I think it is cleaner for these tests to not call cli with string flags but to use client library with `c.Solve()`. There are some tests that check cli directly to make sure it works but if the validated feature doesn't check anything new from the cli then `c.Solve()` is better.